openmdao/core/driver.py

```diff
@@ -1687,6 +1688,289 @@ def _update_result(self, result):
+        # variable.
+        totals = prob.compute_totals(list(of_totals),
+                                     list(des_vars),
+                                     driver_scaling=True)
+
+        grad_f = {inp: totals[obj_name, inp] for inp in des_vars.keys()}
+
+        n = 0
+        for inp in grad_f.keys():
+            n += grad_f[inp].size
+
+        grad_f_vec = np.zeros((n))
+        offset = 0
+        for inp in grad_f.keys():
+            inp_size = grad_f[inp].size
+            grad_f_vec[offset:offset + inp_size] = grad_f[inp]
+            offset += inp_size
+
+        actives = active_cons | active_dvs
+        n_active = np.sum(np.fromiter((len(c['indices']) for c in actives.values()), dtype=int))
+        active_jac_blocks = []
+
+        if n_active > 0:
+            for i, (con_name, active_meta) in enumerate(actives.items()):
+                # If the constraint is a design variable, the constraint gradient
+                # wrt des vars is just an identity matrix sized by the number of
+                # active elements in the design variable.
+                active_idxs = active_meta['indices']
+                if con_name in des_vars.keys():
+                    size = des_vars[con_name]['size']
+                    con_grad = {(con_name, inp): np.eye(size)[active_idxs, ...] if inp == con_name
+                                else np.zeros((size, dv_meta['size']))[active_idxs, ...]
+                                for (inp, dv_meta) in des_vars.items()}
+                else:
+                    con_grad = {(con_name, inp): totals[con_name, inp][active_idxs, ...]
+                                for inp in des_vars.keys()}
+                if use_sparse_solve:
+                    active_jac_blocks.append([sp.csr_matrix(cg) for cg in con_grad.values()])
+                else:
+                    active_jac_blocks.append(list(con_grad.values()))
+
+            if use_sparse_solve:
+                active_cons_mat = sp.block_array(active_jac_blocks)
+            else:
+                active_cons_mat = np.block(active_jac_blocks)
+        else:
+            return {}, {}
+
+        if use_sparse_solve:
+            lstsq_sol = sp.linalg.lsqr(active_cons_mat.T, -grad_f_vec)
+        else:
+            lstsq_sol = np.linalg.lstsq(active_cons_mat.T, -grad_f_vec, rcond=None)
+        multipliers_vec = lstsq_sol[0]
+
+        multipliers = dict()
+        offset = 0
+
+        opts = ['indices', 'ref0', 'ref', 'units']
+        driver_vars = prob.list_driver_vars(out_stream=None,
+                                            desvar_opts=opts + ['flat_indices'],
+                                            cons_opts=opts + ['flat_indices'],
+                                            objs_opts=opts,
+                                            return_format='dict')
+        driver_vars = driver_vars['design_vars'] | driver_vars['constraints']
```

Comment:
is driver_vars used somewhere?

Reply:
Vestige of the way I was doing things previously, fixing.